spring-cloud-scheduler-spi/src/main/java/org/springframework/cloud/scheduler/spi/Scheduler.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+package org.springframework.cloud.scheduler.spi;
+
+import java.util.List;
+
+import org.springframework.cloud.scheduler.spi.core.ScheduleRequest;
+import org.springframework.core.io.Resource;
+
+/**
+ * A {@code Scheduler} is a component that provides a way to register the execution of a
+ * {@link ScheduleRequest} with an underlying scheduler system (Quartz, etc).
+ *
+ * @author Glenn Renfro
+ *
+ */
+
+public interface Scheduler {
+
+	/**
+	 * Registers the {@link ScheduleRequest} to be executed based on the
+	 * cron expression provided.
+	 *
+	 * @param scheduleRequest A request representing a sched-uable
+	 * artifact({@link org.springframework.cloud.deployer.spi.core.AppDefinition},
+	 * the {@link Resource}), schedule properties, and deployment properties.
+	 */
+	void schedule(ScheduleRequest scheduleRequest);
+
+	/**
+	 *  Deletes a schedule that has been created
+	 *
+	 * @param scheduleName the name of the schedule to be removed.
+	 */
+	void unschedule(String scheduleName);
+
+	/**
+	 * List all of the Schedules associated with the provided AppDefinition.
+	 *
+	 * @param taskDefinitionName to retrieve {@link ScheduleInfo}s for a specified taskDefinitionName.
+	 * @return A List of {@link ScheduleInfo}s configured for the provided taskDefinitionName.
+	 */
+	List<ScheduleInfo> list(String taskDefinitionName) ;
```

Comment:
We should indicate what type of exception is thrown in case of failure.  If we have a general `ScheduleException` that take a basic message, then perhaps we can use that .... or come up with another exception type?